boardswarm-cli/src/ui.rs

```diff
@@ -189,68 +150,117 @@ async fn run_ui_internal(
+                                        }
+                                    }
+                                    KeyCode::Char('r') if saw_escape => {
+                                        // Perform power off action
+                                        if let Err(err) = device.change_mode("off").await {
+                                            let err_msg: String =
+                                                format!("Change mode 'off' action failed: {}\r\n", err.code());
+                                                let _ = input_tx.send(Input::UIMessage(err_msg)).await;
+                                        }
+                                        // Perform power on action
+                                        if let Err(err) = device.change_mode("on").await {
+                                            let err_msg: String =
+                                                format!("Change mode 'on' action failed: {}\r\n", err.code());
+                                                let _ = input_tx.send(Input::UIMessage(err_msg)).await;
+                                        }
+                                    }
+
+                                    _ => {
+                                        // Reset escape code
+                                        saw_escape = false;
+
+                                        // Handle Enter key for output console
+                                        if key_code == KeyCode::Enter {
+                                            // Send scroll reset on output console
+                                            let _ = input_tx.send(Input::ScrollReset).await;
+                                        }
+
+                                        // Convert keycode to an equivalent value (ANSI or VT) for input console
+                                        if let Some(val) = convert_keycode_to_term_code(key_code, modifiers) {
+                                            let stream = futures::stream::once(async move { Bytes::from(val) } );
+                                            let _ = console.stream_input(stream).await;
+                                        }
+                                    },
+                                }
                             }
-                            None
-                        }
-                        Input::Up | Input::Down | Input::ScrollReset => {
-                            let _ = input_tx.send(i).await;
-                            None
-                        }
-                        Input::Bytes(data) => Some(data),
-                        Input::UIMessage(_) => None,
+                            Event::Mouse(mouse_event) => match mouse_event.kind {
+                                MouseEventKind::ScrollDown => {
+                                    let _ = input_tx.send(Input::Down).await;
```

Comment:
what if you're the remote console app responds to mouse events? e.g. when running htop remotely?